ddtrace/opentelemetry/tracer.go

```diff
@@ -44,28 +46,48 @@ func (t *oteltracer) Start(ctx context.Context, spanName string, opts ...oteltra
 	if t := ssConfig.Timestamp(); !t.IsZero() {
 		ddopts = append(ddopts, tracer.StartTime(ssConfig.Timestamp()))
 	}
-	for _, attr := range ssConfig.Attributes() {
-		ddopts = append(ddopts, tracer.Tag(toSpecialAttributes(string(attr.Key), attr.Value)))
-	}
 	if k := ssConfig.SpanKind(); k != 0 {
 		ddopts = append(ddopts, tracer.Tag(ext.SpanKind, k.String()))
 	}
-	if opts, ok := spanOptionsFromContext(ctx); ok {
-		ddopts = append(ddopts, opts...)
-	}
 	telemetry.GlobalClient.Count(telemetry.NamespaceTracers, "spans_created", 1.0, telemetryTags, true)
+	ctxOpts, ok := spanOptionsFromContext(ctx)
+	if ok {
+		ddopts = append(ddopts, ctxOpts...)
+	}
+	var cfg ddtrace.StartSpanConfig
+	for _, option := range ctxOpts {
+		option(&cfg)
+	}
+	finalTags := reconcileTags(cfg.Tags, ssConfig.Attributes())
+	// Since there is no way to see if and how the span operation name was set,
+	// we have to record the attributes  locally.
+	// The span operation name will be calculated when it's ended.
 	s := tracer.StartSpan(spanName, ddopts...)
 	os := oteltrace.Span(&span{
 		Span:       s,
 		oteltracer: t,
+		spanKind:   ssConfig.SpanKind(),
+		attributes: finalTags,
 	})
 	// Erase the start span options from the context to prevent them from being propagated to children
 	ctx = context.WithValue(ctx, startOptsKey, nil)
-	// Wrap the span in Opentelemetry and Datadog contexts to propagate span context values
+	// Wrap the span in OpenTelemetry and Datadog contexts to propagate span context values
 	ctx = oteltrace.ContextWithSpan(tracer.ContextWithSpan(ctx, s), os)
 	return ctx, os
 }
 
+func reconcileTags(ddMap map[string]interface{}, otelMap []attribute.KeyValue) map[string]interface{} {
+	result := make(map[string]interface{})
+	for _, attr := range otelMap {
+		result[string(attr.Key)] = attr.Value.AsInterface()
+	}
+
+	for k, v := range ddMap {
+		result[k] = v
+	}
+	return result
+}
+
```

Comment:
I'm not sure I understand why this needs to be a separate function, and why a new map needs to be created. Wouldn't it work to just add the following code to `(t *oteltracer)`, and remove this function entirely:

```
for _, attr := range otelMap {
  cfg.Tags[string(attr.Key)] = attr.Value.AsInterface()
}
```